tsdb/chunkenc/histogram_meta.go

```diff
@@ -284,96 +284,6 @@ type Insert struct {
-//
-// But now we introduce a new bucket layout. (Carefully chosen example where we
-// have a span appended, one unchanged[*], one prepended, and two merge - in
-// that order.)
-//
-// [*] unchanged in terms of which bucket indices they represent. but to achieve
-// that, their offset needs to change if "disrupted" by spans changing ahead of
-// them
-//
-//	                                      \/ this one is "unchanged"
-//	spans      : [  0  ,  3    ]         [1,1]       [    1    ,   4     ]                     [  3  ,   3    ]
-//	bucket idx : [0]   [1]   [2]    3    [4]    5    [6]   [7]   [8]   [9]    10    11    12   [13]  [14]  [15]
-//	raw values    6     3     0           3           0     0     2     4                       5     0     1
-//	deltas        6    -3    -3           3          -3     0     2     2                       1    -5     1
-//	delta mods:                          / \                     / \                                       / \
-//
-// Note for histograms with delta-encoded buckets: Whenever any new buckets are
-// introduced, the subsequent "old" bucket needs to readjust its delta to the
-// new base of 0. Thus, for the caller who wants to transform the set of
-// original deltas to a new set of deltas to match a new span layout that adds
-// buckets, we simply need to generate a list of inserts.
-//
-// Note: Within expandSpansForward we don't have to worry about the changes to the
-// spans themselves, thanks to the iterators we get to work with the more useful
-// bucket indices (which of course directly correspond to the buckets we have to
-// adjust).
-func expandSpansForward(a, b []histogram.Span) (forward []Insert, ok bool) {
-	ai := newBucketIterator(a)
-	bi := newBucketIterator(b)
-
-	var inserts []Insert
-
-	// When inter.num becomes > 0, this becomes a valid insert that should
-	// be yielded when we finish a streak of new buckets.
-	var inter Insert
-
-	av, aOK := ai.Next()
-	bv, bOK := bi.Next()
-loop:
-	for {
-		switch {
-		case aOK && bOK:
-			switch {
-			case av == bv: // Both have an identical value. move on!
-				// Finish WIP insert and reset.
-				if inter.num > 0 {
-					inserts = append(inserts, inter)
-				}
-				inter.num = 0
-				av, aOK = ai.Next()
-				bv, bOK = bi.Next()
-				inter.pos++
-			case av < bv: // b misses a value that is in a.
-				return inserts, false
-			case av > bv: // a misses a value that is in b. Forward b and recompare.
-				inter.num++
-				bv, bOK = bi.Next()
-			}
-		case aOK && !bOK: // b misses a value that is in a.
-			return inserts, false
-		case !aOK && bOK: // a misses a value that is in b. Forward b and recompare.
-			inter.num++
-			bv, bOK = bi.Next()
-		default: // Both iterators ran out. We're done.
-			if inter.num > 0 {
-				inserts = append(inserts, inter)
-			}
-			break loop
-		}
-	}
-
-	return inserts, true
-}
-
 // expandSpansBothWays is similar to expandSpansForward, but now b may also
```

Comment:
Reference to now deleted function expandSpansForward. There are a few other references to expandSpansForward elsewhere in the code as well

Reply:
done